Define action displayed names in DataView

vickyliu-go4it opened the pull request:
We changed to define action displayed names in DataView instead of each
individual views. This avoids the redifinition and inconsistencies of
action displayed names in different views.

Bug: http://b/205676296

beckerhe commented on src/DataViews/include/DataViews/DataView.h:
```diff
@@ -104,6 +104,35 @@ class DataView {
   [[nodiscard]] virtual bool ResetOnRefresh() const { return true; }
 
+  // Hooking related actions
+  static const std::string kMenuActionLoadSymbols;
```
Would it work to make these `constexpr std::string_view`s and completely define them in the header? These `const std::string` always felt wrong to me.

beckerhe replied:
Sorry for bringing this up so late. :see_no_evil:

beckerhe reviewed (approved):
Hey Vicky, thanks a lot for the change. I added one comment which should be fixed. Otherwise it looks good to me.

But I wanted to clarify myself. When I said I preferred string_view I meant only the definition of the constants. Now you have also changed all the interfaces to string_view and that could be a problem in the future because it doesn't all dynamic menu entries. If you keep string_view there, all menu entry strings need to come from some static storage (like constants). That's not a problem right now, but could be in the future when we want to introduce dynamic entries.

beckerhe commented on src/DataViews/CallstackDataView.cpp:
```diff
@@ -129,21 +124,21 @@ std::vector<std::vector<std::string>> CallstackDataView::GetContextMenuWithGroup
 
-  std::vector<std::string> action_group;
+  std::vector<std::string_view> action_group;
   if (enable_load) action_group.emplace_back(kMenuActionLoadSymbols);
   if (enable_select) action_group.emplace_back(kMenuActionSelect);
   if (enable_unselect) action_group.emplace_back(kMenuActionUnselect);
   if (enable_disassembly) action_group.emplace_back(kMenuActionDisassembly);
   if (enable_source_code) action_group.emplace_back(kMenuActionSourceCode);
```
If you decided to move from string_view back to string in the function signature, then you would need to make explicit constructions of strings here.

vickyliu-go4it replied:
Done! Thanks for the suggestion!

vickyliu-go4it commented:
> Hey Vicky, thanks a lot for the change. I added one comment which should be fixed. Otherwise it looks good to me.
> 
> But I wanted to clarify myself. When I said I preferred string_view I meant only the definition of the constants. Now you have also changed all the interfaces to string_view and that could be a problem in the future because it doesn't all dynamic menu entries. If you keep string_view there, all menu entry strings need to come from some static storage (like constants). That's not a problem right now, but could be in the future when we want to introduce dynamic entries.

Hi Henning, sorry I misunderstand you suggestion. I found I need to convert `std::string_view` back to `std::string` each time when I use the action name in `GetContextMenuWithGrouping`.

I reverted the previous fixup and changed to:
* only use string_view for the definition of constants
* explicit construct to std::string in `GetContextMenuWithGrouping`.

vickyliu-go4it commented:
> Hey Vicky, thanks a lot for the change. I added one comment which should be fixed. Otherwise it looks good to me.
> 
> But I wanted to clarify myself. When I said I preferred string_view I meant only the definition of the constants. Now you have also changed all the interfaces to string_view and that could be a problem in the future because it doesn't all dynamic menu entries. If you keep string_view there, all menu entry strings need to come from some static storage (like constants). That's not a problem right now, but could be in the future when we want to introduce dynamic entries.

Hi Henning, sorry I misunderstand you suggestion. I found I need to convert `std::string_view` back to `std::string` each time when I use the action name in `GetContextMenuWithGrouping`.

I reverted the previous fixup and changed to:
* only use string_view for the definition of constants
* explicit construct to std::string in `GetContextMenuWithGrouping`.

vickyliu-go4it commented on src/OrbitQt/orbittreeview.cpp:
```diff
@@ -214,8 +214,8 @@ void OrbitTreeView::ShowContextMenu(const QPoint& pos) {
 
-      for (const std::string& action_name : menu_with_grouping[i]) {
-        actions.push_back(std::make_unique<QAction>(QString::fromStdString(action_name)));
+      for (std::string_view action_name : menu_with_grouping[i]) {
+        actions.push_back(std::make_unique<QAction>(action_name.data()));
```
Thanks for the explanation! I changed to only use std::string_view when defining the constants
